### PR TITLE
Add error logging middleware

### PR DESCRIPTION
**Pull request by srallen**
Package: app-project

This is an experiment to add middleware to MST store actions to wrap them in a try/catch and then if they fail, to abort the action, and console log and log to sentry. 

The reason for this is that I would've expected to see some errors in the Sentry logs by now and there are none, which I don't believe. I'm wondering if the current setup isn't capturing errors in the MST stores. 

How can we best test this to actually trigger a log to go to Sentry?

If this works, I'd like to create a logger library so then it can be used in the MST stores for the classifier since I don't think those are getting to Sentry either.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

**Comment by rogerhutchings** [comment]
cc #962

File: packages/app-project/stores/Store.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import { getEnv, types } from 'mobx-state-tree'
+import { addMiddleware, getEnv, types } from 'mobx-state-tree'
+import { logNodeError } from '../src/helpers/logger'
```
Comment (rogerhutchings) [inline packages/app-project/stores/Store.js]: fwiw I think I named this function poorly

**Comment by rogerhutchings** [comment]
I _think_ I'm okay with this. I think there are some unexpected side effects that we'll find by aborting the existing call, buuuut no worse than crashing the app which is what will probably happen anyway.

**Comment by srallen** [comment]
@rogerhutchings Only if we have a plan on how to test this on staging. I'm still not sure how to go about that.

**Comment by rogerhutchings** [comment]
Assuming that Sentry is set up on staging...

- Add a new page with a weird URL that can only be routed to from staging (https://frontend.preview.zooniverse.org/projects/test/project/foobar)
- This page has a button that triggers an error in MST
- We merge it, it gets deployed to staging
- Hit the magic error button
- See if we get a Sentry error

File: packages/app-project/stores/Store.js (hunk 1 above)
Comment (eatyourgreens) [reply packages/app-project/stores/Store.js:2]: I’ve renamed it `logToSentry` in #2044.

**Pull request merged**